SAST Report on function ‘void icMemDump(std::string&, void*, size_t)’:

## Maintainer Report

Mon Jan  5 02:26:18 AM UTC 2026

Issue: SAST Report on function ‘void icMemDump(std::string&, void*, size_t)’:
SAST Log: https://github.com/InternationalColorConsortium/iccDEV/actions/runs/20702517319/job/59427126296#step:7:409


## SAST log


```
  /home/runner/work/iccDEV/iccDEV/IccProfLib/IccUtil.cpp: In function ‘void icMemDump(std::string&, void*, size_t)’:
  /home/runner/work/iccDEV/iccDEV/IccProfLib/IccUtil.cpp:983:14: warning: ‘char* __builtin_strncpy(char*, const char*, long unsigned int)’ output may be truncated copying 9 bytes from a string of length 9 [-Wstringop-truncation]
    983 |       strncpy(buf, num, 9);
        |              ^
  /home/runner/work/iccDEV/iccDEV/IccProfLib/IccUtil.cpp:987:12: warning: ‘char* __builtin_strncpy(char*, const char*, long unsigned int)’ output may be truncated copying 2 bytes from a string of length 9 [-Wstringop-truncation]
    987 |     strncpy(buf+10+j*3, num, 2);
        |    
```

## Request

Please consider refactoring and/or adding comments to the code if this is the intended behavior when time permits. If this is the intended behavior please Label as, OOS, wontfix and Close.

Thank You

--- a/IccProfLib/IccUtil.cpp
+++ b/IccProfLib/IccUtil.cpp
@@ -980,11 +980,11 @@ void icMemDump(std::string &sDump, void *pBuf, size_t nNum)
       buf[77] = '\n';
       buf[78] = '\0';
       snprintf(num, numSize, "%08X:", uint32_t(i));    // NOTE - this formatting will have to change if we want to move beyond 4 Gig.
-      strncpy(buf, num, 9);
+      memcpy(buf, num, 9); // 8 hex digits, plus ':', and no terminating NULL
     }
 
     snprintf(num, numSize, "%02X", *pData);
-    strncpy(buf+10+j*3, num, 2);
+    memcpy(buf+10+j*3, num, 2);    // 2 hex digits, no terminating NULL
 
     c=*pData;
     if (!isprint(c) || c > 126)

